### PR TITLE
Add pre-run-script to the test workflow

### PR DESCRIPTION
**Pull request by weiiwang01**

### CODE REVIEW

**Review by arturo-seijas — APPROVED** [review]
LGTM

**Review by cbartz — APPROVED** [review]
LGTM

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -14,6 +14,9 @@ on:
         type: boolean
         description: Whether to use self-hosted runners to run the jobs.
         default: true
+      pre-run-script:
+        description: Path to the bash script to be run before the integration tests
```
Comment (jdkandersson) [inline .github/workflows/test.yaml:18]: Should this say "integration" test? Also, should this be documented in the readme?

Comment (weiiwang01) [reply .github/workflows/test.yaml:18]: I think this is copied from the input definition of the integration test workflow. I will fix the in the next pull request, thanks!

**Pull request merged**